test/test_prototype_datasets_builtin.py

```diff
@@ -78,23 +79,33 @@ def test_sample(self, dataset_mock, config):
         if not sample:
             raise AssertionError("Sample dictionary is empty.")
 
+        list(iterator)  # Cleanups and closing streams in buffers
+
     @parametrize_dataset_mocks(DATASET_MOCKS)
     def test_num_samples(self, dataset_mock, config):
         dataset, mock_info = dataset_mock.load(config)
-
         assert len(list(dataset)) == mock_info["num_samples"]
 
     @parametrize_dataset_mocks(DATASET_MOCKS)
     def test_no_vanilla_tensors(self, dataset_mock, config):
+        StreamWrapper.session_streams = {}
         dataset, _ = dataset_mock.load(config)
 
-        vanilla_tensors = {key for key, value in next(iter(dataset)).items() if type(value) is torch.Tensor}
+        iterator = iter(dataset)
+        one_element = next(iterator)
+
+        vanilla_tensors = {key for key, value in one_element.items() if type(value) is torch.Tensor}
         if vanilla_tensors:
             raise AssertionError(
                 f"The values of key(s) "
                 f"{sequence_to_str(sorted(vanilla_tensors), separate_last='and ')} contained vanilla tensors."
             )
 
+        list(iterator)  # Cleanups and closing streams in buffers
+
+        if len(StreamWrapper.session_streams) > 0:
+            raise Exception(StreamWrapper.session_streams)
```

Comment:
Could you explain what this does? Is `StreamWrapper.session_streams` just a counter for open streams? If yes, why are we only testing this here and not in the other tests? If this is something we should check in general, we can use a decorator like

```py
def check_unclosed_streams(test_fn):
    @functools.wraps(test_fn)
    def wrapper(*args, **kwargs):
        if len(StreamWrapper.session_streams) > 0:
            raise pytest.UsageError("Some previous test didn't clean up")
        
        test_fn(*args, **kwargs)
        
        if len(StreamWrapper.session_streams) > 0:
            raise Assertion("This test didn't clean up")
        
    return wrapper
```